File: 172.factorial-trailing-zeroes.js
/*
 * @lc app=leetcode id=172 lang=javascript
 *
 * [172] Factorial Trailing Zeroes
 */

// @lc code=start
/**
 * @param {number} n
 * @return {number}
 */
var trailingZeroes = function (n) {
  let res = 0;
  while (n > 0) {
    n = Math.trunc(n / 5);
    res += n;
  }
  return res;
};
// @lc code=end

File: 172.factorial-trailing-zeroes.go
/*
 * @lc app=leetcode id=172 lang=golang
 *
 * [172] Factorial Trailing Zeroes
 */

package leetcode

// @lc code=start
func trailingZeroes(n int) int {
	res := 0
	for n > 0 {
		res += n / 5
		n /= 5
	}
	return res
}

// @lc code=end
